cobra/flux_analysis/summary.py

```diff
@@ -20,16 +22,145 @@
 LOGGER = logging.getLogger(__name__)
 
 
-def metabolite_summary(met, solution=None, threshold=0.01, fva=False,
-                       names=False, floatfmt='.3g'):
+class Summary(object):
+    """Class definition for a Summary object.
+
+    Parameters
+    ----------
+    solution : cobra.Solution or None
+        A previously solved model solution to use for generating the
+        summary. If none provided (default), the summary method will
+        resolve the model. Note that the solution object must match the
+        model, i.e., changes to the model such as changed bounds,
+        added or removed reactions are not taken into account by this
+        method.
+    threshold : float
+        Threshold below which fluxes are not reported.
+    fva : pandas.DataFrame, float or None
+        Whether or not to include flux variability analysis in the output.
+        If given, fva should either be a previous FVA solution matching
+        the model or a float between 0 and 1 representing the
+        fraction of the optimum objective to be searched.
+    names : bool
+        Emit reaction and metabolite names rather than identifiers (default
+        False).
+    floatfmt : string
+        Format string for floats (default '.3g').
+
     """
-    Print a summary of the production and consumption fluxes.
 
-    This method requires the model for which this metabolite is a part
-    to be solved.
+    def __init__(self, solution, threshold, fva, names, floatfmt):
+        self.solution = solution
+        self.threshold = threshold
+        self.fva = fva
+        self.names = names
+        self.floatfmt = floatfmt
+
+    def _generate():
```

Comment:
- `self` argument missing, i.e., `_generate(self):`
- perhaps mention that this is abstract and is implemented by the respective summaries

Reply:
Yes will add the `self` argument and also update the docstrings as well.